vast/opensetAlgos/WPL.py

```diff
@@ -0,0 +1,172 @@
+the point (input) and the center (Prototype). So, distance is a vector with 'd' dimension. 
+Distance is not a scalar. The probability of each class is the minimum Weibull probability of all dimensions.
+
+In the training time, if a Weibll (class, dimension) has 5 or more unique values, maximum likelihood 
+estimator (MLE)  or maximum a posterior (MAP) can be used to estimate the Weibull parameter. 
+If a Weibll (class, dimension) has less than 5 unique values, instead of estimating, it uses 
+the default parameter for creating Weibull. The default parameters are the default scale and default shape.
+
+
+WPL has 4 arguments: tail size, distance multiplier, default_scale, default shape
+
+The main difference between WPL and OpenMax is how they compute the distance. 
+Therefore, the number of Weibull is different between WPL and OpenMax. Another difference 
+is how they are training Weibull when the number of unique values is less than 5.
+
+
+"""
+
+import torch
+import itertools
+from ..tools import pairwisedistances
+from ..DistributionModels import weibull
+from typing import Iterator, Tuple, List, Dict, OrderedDict
+
+
+def WPL_Params(parser):
+    WPL_params = parser.add_argument_group("WPL params")
+    WPL_params.add_argument(
+        "--tailsize",
+        nargs="+",
+        type=float,
+        default=[1.0],
+        help="tail size to use default: %(default)s",
+    )
+    WPL_params.add_argument(
+        "--distance_multiplier",
+        nargs="+",
+        type=float,
+        default=[1.0],
+        help="distance multiplier to use default: %(default)s",
+    )
+    WPL_params.add_argument(
+        "--default_scale",
+        nargs="+",
+        type=float,
+        default=[1.0],
+        help="Weibull scale to use when the number of uniqe element is less than 5 default: %(default)s",
+    )
+    WPL_params.add_argument(
+        "--default_shape",
+        nargs="+",
+        type=float,
+        default=[0.1],
+        help="Weibull shape to use when the number of uniqe element is less than 5 default:  %(default)s",
+    )
+    return parser, dict(
+        group_parser=WPL_params,
+        param_names=("tailsize", "distance_multiplier", "default_scale", "default_shape"),
+        param_id_string="TS_{}_DM_{:.4f}_SC_{:.4f}_SH_{:.4f}",
+    )
+
+
+def fit_high(distances, distance_multiplier, tailsize, default_shape, default_scale):
+    distances = torch.unique(distances)
+    if tailsize <= 1:
+        tailsize = min(tailsize * distances.shape[1], distances.shape[1])
+    tailsize = int(min(tailsize, distances.shape[1]))
+    mr = weibull.weibull()
+    if distances.shape[1] < 5:
+        pass
+        mr.sign = 1
+        mr.wbFits = torch.zeros(1, 2)
+        mr.wbFits[0, 1] = default_scale
+        mr.wbFits[0, 0] = default_shape
+        mr._ = torch.Tensor(0.0) # translate Amount
+        mr.smallScoreTensor =  torch.Tensor(0.0)#  small Score
```

Comment:
You could just create this as done at https://github.com/Vastlab/vast/blob/460480fc3bf624d1ffee70be5ec446bf6f6ceee6/vast/opensetAlgos/multimodal_openmax.py#L119